tests/integration/submissionworker/test_submission_worker.py

```diff
@@ -0,0 +1,512 @@
+        f.write("file_content")
+
+        self.zipped = zipfile.ZipFile(join(self.BASE_TEMP_DIR, 'test_zip.zip'), 'w')
+        self.zipped.write(join(self.BASE_TEMP_DIR, 'dummy_input.txt'), 'dummy_input.txt')
+        self.zipped.close()
+        file = open(join(self.BASE_TEMP_DIR, 'test_zip.zip'), 'rb')
+        self.z = SimpleUploadedFile(join(self.BASE_TEMP_DIR, 'test_zip.zip'), file.read(), content_type='application/zip')
+
+        self.challenge = Challenge.objects.create(
+            title='Test Challenge',
+            description='Description for test challenge',
+            terms_and_conditions='Terms and conditions for test challenge',
+            submission_guidelines='Submission guidelines for test challenge',
+            creator=self.challenge_host_team,
+            start_date=timezone.now(),
+            end_date=timezone.now() + timedelta(days=1),
+            published=True,
+            evaluation_script=self.z,
+            approved_by_admin=True,
+            enable_forum=True,
+            anonymous_leaderboard=False)
+
+        try:
+            os.makedirs("/tmp/evalai")
+        except OSError:
+            pass
+
+        with self.settings(MEDIA_ROOT='/tmp/evalai'):
+            self.challenge_phase = ChallengePhase.objects.create(
+                name='Challenge Phase',
+                description='Description for Challenge Phase',
+                leaderboard_public=False,
+                is_public=True,
+                start_date=timezone.now(),
+                end_date=timezone.now() + timedelta(days=1),
+                challenge=self.challenge,
+                test_annotation=SimpleUploadedFile('test_sample_file.txt',
+                                                   b'Dummy file content',
+                                                   content_type='text/plain')
+            )
+
+        self.url = reverse_lazy('jobs:challenge_submission',
+                                kwargs={'challenge_id': self.challenge.pk,
+                                        'challenge_phase_id': self.challenge_phase.pk})
+
+        self.client.force_authenticate(user=self.user1)
+
+        self.input_file = SimpleUploadedFile(
+            "test_file.txt", b"file_content", content_type="text/plain")
+
+        self.submission = Submission.objects.create(
+            participant_team=self.participant_team,
+            challenge_phase=self.challenge_phase,
+            created_by=self.challenge_host_team.created_by,
+            status="submitted",
+            input_file=self.input_file,
+            method_name="Test Method",
+            method_description="Test Description",
+            project_url="http://testserver/",
+            publication_url="http://testserver/",
+            is_public=True,
+        )
+
+    def tearDown(self):
+        try:
+            shutil.rmtree('/tmp/evalai')
+        except OSError:
+            pass
+
+
+class NormalTestClass(BaseAPITestClass):
```

Comment:
Can we rename the test class to something more relevant to the test cases?

Reply:
I've refactored the code. PTAL.